lips/lip-10.md

```diff
@@ -0,0 +1,221 @@
+* **Payment endpoint** to perform actual money transfer.
+
+Common considerations:
+
+* For both endpoints, all communications are signed via the same mechanism as for other off-chain communications, see the [Payloads section of lip-1](https://github.com/libra/lip/blob/master/lips/lip-1.mdx#payloads) for details. As such, all vasp-2-vasp communications are authenticated.
+
+* All Libra ID endpoints are nested under `pay.<version>` URI path:
+
+     `https://<hostname>:<port>/pay.<protocol_version>/<endpoint>`
+
+* All Libra ID endpoints use the same `base_url` and same `compliance_public_key` for authentication as other off-chain APIs.
+
+* All Libra ID endpoints are HTTP POST endpoints, they accept JSON-encoded requests and return JSON-encoded responses. Other encoding methods can be considered in the future, `Content-Type` and `Accept` HTTP headers can be used to define request/response formats.
+
+## Info endpoint
+
+```
+POST https://<hostname>:<port>/pay.<protocol_version>/info
+```
+
+**Info endpoint** is used by the **Sender** to request information about the **Receiver** before submitting an actual payment.
+
+Info endpoint takes both sender and receiver Libra IDs as arguments. Receiver VASP might decide to share more(or less) information based on sender ID, for example if the wallet supports some kind of contact list, they might decide to share more information if the sender is in the contact list of the receiver.
+
+**Request:**
+```json
+{
+   "sender": "<User Object>",
+   "receiver_id": "<Receiver Libra ID>"
+}
+```
+
+| Field 	   | Type 	     | Required? 	| Description 	           |
+|-------	   |------	     |-----------	|-------------	           |
+|sender        |[User Object](https://github.com/libra/lip/blob/master/lips/lip-10.md#user-object)  | Y            | Information about the sender |
+|receiver_id   |str          | Y            | Libra ID of the receiver |
+
+**Response:**
+```json
+{
+   "receiver": "<User Object>"
+}
+```
+
+| Field 	   | Type 	     | Required? 	| Description 	             |
+|-------	   |------	     |-----------	|-------------	             |
+|receiver      |[User Object](https://github.com/libra/lip/blob/master/lips/lip-10.md#user-object)  | Y            | Information about the receiver |
+
+* Receiver Libra ID domain must belong to the Receiver VASP. Otherwise, Receiver VASP must return the `400` error code.
+* Sender Libra ID must have a Libra ID domain that belongs to Sender VASP. If this is not the case, the Receiver VASP should return the `400` error code 
+* Receiver VASP may optionally choose how much information to share based on the sender Libra ID. If Receiver VASP decides to not share data, it should return an empty `User Object` with only `libra_id` field set
+* If user is not found, the VASP should return error code `400`  
+
+## Pay endpoint
+
+```
+POST https://<hostname>:<port>/pay.<protocol_version>/pay
+```
+
+**Pay endpoint** is used when a user is willing to make a money transfer.
```

Comment:
Why are we doing this off chain? It feels like I can embed some metadata in the transaction and receiving vasp  can use the lookup service... this feels a little unnatural.